kubespawner/spawner.py

```diff
@@ -1640,6 +1790,9 @@ def _start(self):
         if events:
             self._last_event = events[-1].metadata.uid
 
+        if not self._has_namespace(self.namespace):
```

Comment:
avoid calling this in the default case:

```suggestion
        if self.namespace_name_template and not self._has_namespace(self.namespace):
```